clustered/client/src/test/java/org/ehcache/clustered/client/internal/UnitTestConnectionService.java

```diff
@@ -26,29 +28,61 @@
 import org.terracotta.connection.ConnectionService;
 import org.terracotta.consensus.entity.CoordinationServerEntityService;
 import org.terracotta.consensus.entity.client.ClientCoordinationEntityService;
+import org.terracotta.offheapresource.OffHeapResourcesConfiguration;
+import org.terracotta.offheapresource.OffHeapResourcesProvider;
+import org.terracotta.offheapresource.config.MemoryUnit;
+import org.terracotta.offheapresource.config.OffheapResourcesType;
+import org.terracotta.offheapresource.config.ResourceType;
 import org.terracotta.passthrough.PassthroughServer;
 
 public class UnitTestConnectionService implements ConnectionService {
 
-  public static PassthroughServer createServer() {
+  private static PassthroughServer createServer() throws Exception {
     PassthroughServer newServer = new PassthroughServer(true);
     newServer.registerServerEntityService(new EhcacheServerEntityService());
     newServer.registerClientEntityService(new EhcacheClientEntityService());
     newServer.registerServerEntityService(new CoordinationServerEntityService());
     newServer.registerClientEntityService(new ClientCoordinationEntityService());
+
+    /*
+     * Construct an off-heap resource configuration and register an OffHeapResourcesProvider.
+     */
+    OffheapResourcesType resources = new OffheapResourcesType();
+    List<ResourceType> resourcesList = resources.getResource();
+    resourcesList.add(defineResource("defaultResource", MemoryUnit.MB, 128L));
+    resourcesList.add(defineResource("primary-server-resource", MemoryUnit.MB, 64L));
+    resourcesList.add(defineResource("secondary-server-resource", MemoryUnit.MB, 64L));
+    OffHeapResourcesConfiguration ohrpConfig = new OffHeapResourcesConfiguration(resources);
+    newServer.registerServiceProvider(new OffHeapResourcesProvider(), ohrpConfig);
+
     newServer.start();
     return newServer;
   }
 
-  private static PassthroughServer server = createServer();
+  private static ResourceType defineResource(String name, MemoryUnit unit, long size) {
+    final ResourceType resource = new ResourceType();
+    resource.setName(name);
+    resource.setUnit(unit);
+    resource.setValue(BigInteger.valueOf(size));
+    return resource;
+  }
+
+  private static PassthroughServer server;
+  static {
```

Comment:
This class and its usage pattern existed before this PR.  It does make sense to alter the usage pattern for this and, if it makes sense (from an effort point of view) for me to change this for the unit tests I'm working on for this commit stream, I'll go ahead and do it then.. Otherwise I'll open an issue for it.